core/arch/arm/kernel/generic_entry_a64.S

```diff
@@ -47,7 +47,8 @@
 		adr	x2, stack_tmp_stride
 		ldr	w1, [x2]
 		mul	x2, x0, x1
-		adr	x1, stack_tmp
+		adrp	x1, stack_tmp
+		add	x1, x1, :lo12:stack_tmp
```

Comment:
Yeah I guess we don't care about the slight speed difference or even whether the code is relocatable (but that's an interesting argument in some situations obviously).
I'm OK with the adrp/add version.